### DISCUSSION TITLE
Proposal duplicati

### DISCUSSION THREAD

**Discussion by Jin33334** [tteck/Proxmox#320] [Ideas]
Hi,

first of all nice work here. Well done.

I have found every programm i wan implement here, just missing one.
Duplacati, if you think this program is worth. You can can implment it in your summary.

https://duplicati.readthedocs.io/en/latest/02-installation/#installing-duplicati-on-linux

Thank you

**Comment by tteck** [comment]
Curious, what would be the use case for Duplacati with Proxmox?

**Reply by Jin33334** [reply]
You right, you have a point. It can be solved more elegant way.

In my case i go back nas with nextcloud server is in asia. I was planing use duplacati for back up the data in my home back in europe.

**Reply by tteck** [reply]
Ok, here's my backup schema using rclone
Daily at 03:30 Proxmox performs a full backup of every VM/CT to a external SSD drive with 7 backup retention
Sunday at 01:00 Proxmox performs a full backup of every VM/CT to a different external SSD drive with 1 backup retention 
Sunday at 02:00 rclone uploads the Sunday backups to Google Drive
Hourly rclone syncs HA's data folder with Google Drive

**Reply by c0bnut** [reply]
Duplicati is proper backup solution and more than just a sync program, it provides de-duplication, compression, encryption, incremental backups and a nice web GUI to manage it all. Although it looks like BorgBackup might be a better and more reliable solution however it doesn't have a nice GUI to set things up.
EDIT: Looks like kopia GUI is what I'm looking for, will set up another thread to request.

**Reply by guarddog13** [reply]
I need to learn how to do auto backups like this. I can obviously do backup/snapshot, but can't figure out how to automate it or send it to say Google Drive. With this I wouldn't have a need for Google auto backup in HA either.

**Reply by tteck** [reply]
@guarddog13 Now, all I'm doing is backing up every 4hrs with a retention of 24 (4 days)
![Screenshot 2022-10-01 8 39 04 AM](https://user-images.githubusercontent.com/21342670/193410147-25eff3ff-109b-4fcb-bab2-8c52dbddbfe7.png)

**Reply by guarddog13** [reply]
Ohhhh and I just bought a NAS so that'll help. My Ubuntu VM backup is like 25 GBs and I have no clue why it's so large. I record short clips on my Agent DVR that get sent to an HA notification but those get deleted regularly. The only I find that's large on there is /dev/mapper/ubuntu--vg-ubuntu--lv which using 19GB of it's 100GB free.

**Reply by tteck** [reply]
@guarddog13, I backup to a 512GB NVME external USB drive formatted fat32, so if need be, I can plug the drive into any computer and access the backups. Backing up 11 CT's take roughly 3min. Here's what the drive looks like with 264 backups on it (under 200GB)
![Screenshot 2022-10-01 10 53 23 AM](https://user-images.githubusercontent.com/21342670/193415492-6773ee59-969c-4a4a-bd3d-c4d033ec8195.png)

**Comment by guarddog13** [comment]
I have 7 LXCs, one OPNsense VM and one Ubuntu VM. The Ubuntu VM runs both AgentDVR and Deepstack on a GPU. It was much easier to pass through a GPU to a VM and when I was running them on the Ubuntu LXC with docker inside it they were fairly buggy. AgentDVR did not like being inside of two containers at all. OPNsense made sense to run as a VM and it doesn't have a lot of overhead. All of that is about 56GBs. I'm getting a 2TB NAS...so figure a day worth of backups...like I did on HA...and then can run a small plex server on there for a later project.

**Comment by Deekerman** [comment]
Would this allow you to back up your desktop to your server? that would be neat.
